Objects/obmalloc.c

```diff
@@ -1231,6 +1231,24 @@ _PyObject_XDecRefDelayed(PyObject *ptr)
 }
 #endif
 
+#ifdef Py_GIL_DISABLED
+void
+_PyObject_XSetRefDelayed(PyObject **ptr, PyObject *value)
+{
+    PyObject *old = *ptr;
+    FT_ATOMIC_STORE_PTR_RELEASE(*ptr, value);
+    if (old == NULL) {
+        return;
+    }
+    if (!_Py_IsImmortal(old)) {
+         _PyObject_XDecRefDelayed(old);
+    }
+    else {
+         Py_DECREF(old);
```

Comment:
if the object is immortal then decref has no effect

Reply:
https://github.com/python/cpython/blob/598aa7cc98bc1b39f10ec41decddd8dd88799fe1/Include/refcount.h#L375-L378

Reply:
But let's remove it, unitil we need this stat explicitly.